icu4c/source/i18n/collationdatabuilder.cpp

```diff
@@ -261,6 +261,10 @@ DataBuilderCollationIterator::getCE32FromBuilderData(uint32_t ce32, UErrorCode &
         return utrie2_get32(builder.trie, jamo);
     } else {
         ConditionalCE32 *cond = builder.getConditionalCE32ForCE32(ce32);
+        if (cond == nullptr) {
```

Comment:
1. This whole function needs, as the very first thing, to check its incoming UErrorCode, and return immediately if failure.
2.  The fuzzer failure might disappear if this were done (but this would need to be verified)

Reply:
I added that check, as 
```
 uint32_t
 DataBuilderCollationIterator::getCE32FromBuilderData(uint32_t ce32, UErrorCode &errorCode) {
+    if (U_FAILURE(errorCode)) {
+        return 0;
+    }
     U_ASSERT(Collation::hasCE32Tag(ce32, Collation::BUILDER_DATA_TAG));

```
but still have the problem